lib/route.js

```diff
@@ -268,6 +269,10 @@ function buildRouting (options) {
               if (typeof func !== 'function') {
                 throw new FST_ERR_HOOK_INVALID_HANDLER(hook, Object.prototype.toString.call(func))
               }
+
+              if (func.constructor.name === 'AsyncFunction' && func.length === 3) {
```

Comment:
not all hooks have the same parameters:

 'onTimeout' - 3
  'onRequest' - 3
  'preParsing' - 4
  'preValidation' - 3
  'preSerialization' - 4
  'preHandler' - 3
  'onSend' - 4
  'onResponse' - 3
  'onError' - 4
  'onRequestAbort - 2

Reply:
Following your suggestion and the check done [here](https://github.com/fastify/fastify/blob/7d97dcdf2643bcfbdcf4850867139db8189b82e2/fastify.js#L612) I came up with a small refactor. Do you think it's worth it or should I revert it?